src/main/webapp/app/course/manage/course-for-dashboard-dto.ts

```diff
@@ -0,0 +1,14 @@
+import { Course } from 'app/entities/course.model';
+import { CourseScores } from 'app/course/course-scores/course-scores';
+import { Result } from 'app/entities/result.model';
+import { ExerciseType } from 'app/entities/exercise.model';
+
+export class CourseForDashboardDTO {
+    course: Course;
+
+    totalScores: CourseScores;
+    scoresPerExerciseType: { [key in ExerciseType]: CourseScores };
```

Comment:
can you also use a proper Map type here?

Reply:
I think you even have defined a data type already that you could use

Reply:
The issue is that this `CourseForDashboardDTO` represents the data returned from the server and the deserialized JSON will not contain an actual Map. Typescript won't even complain about this and just lets me type this as `ScoresPerExerciseType` here but referencing the data in the component with `scoresPerExerciseType.get(...)` will not work then because the data is not actually a Map. I need to convert it, which is what I do in the `CourseManagementService#saveScoresInStorage()` method. So the `{ [key in ExerciseType]: CourseScores }` is the correct type. I added comments in all places where this issue comes up.